Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+@class FIRAggregateQuery;
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(AggregateQuerySnapshot)
+@interface FIRAggregateQuerySnapshot : NSObject
+/** :nodoc: */
+- (id)init __attribute__((unavailable("FIRAggregateQuerySnapshot cannot be created directly.")));
+
+@property(nonatomic, strong, readonly) FIRAggregateQuery *query;
+
+- (nullable NSNumber*)count;
```

Comment:
I'm not sure if this is a good idea or not, but have you considered always returning count for a given aggregate query?

Reply:
Yes, we have considered always returning count for a given aggregate query. It is _not_ something that we will support. We decided that the client must explicitly request the count. So in a future world where customers can request various aggregations, if they don't request count then `count` will return `nil`.